crates/forge/src/shared_cache.rs

```diff
@@ -0,0 +1,64 @@
+pub const CACHE_DIR: &str = ".snfoundry_cache";
+pub const PREV_TESTS_FAILED: &str = ".prev_tests_failed";
+
+use anyhow::{Ok, Result};
+use camino::Utf8PathBuf;
+use scarb_metadata::MetadataCommand;
+use std::fs::{self, File};
+use std::io::{BufRead, BufReader, BufWriter, Write};
+
+use crate::test_case_summary::TestCaseSummary;
+
+pub fn get_cached_failed_tests_names(workspace_root: &Utf8PathBuf) -> Result<Option<Vec<String>>> {
+    let tests_failed_path = get_cache_dir(workspace_root)?.join(PREV_TESTS_FAILED);
+    if !tests_failed_path.exists() {
+        return Ok(None);
+    }
+
+    let file = File::open(tests_failed_path)?;
+    let buf: BufReader<File> = BufReader::new(file);
+    let tests: Vec<String> = buf
+        .lines()
+        .map(|l| l.expect("Could not parse line"))
+        .collect();
+    if tests.is_empty() {
+        return Ok(None);
+    }
+    Ok(Some(tests))
+}
+
+fn get_cache_dir(workspace_root: &Utf8PathBuf) -> Result<Utf8PathBuf> {
+    Ok(workspace_root.join(CACHE_DIR))
+}
```

Comment:
Isn't it better to just pass the `workspace_root.join(CACHE_DIR)` to every function?